### DISCUSSION TITLE
Cannot install packages from requirements.txt on Windows

### DISCUSSION THREAD

**Discussion by marybj-ocwcog** [UberPython/web-apis-with-python#26] [Q&A]
I have specified a new conda environment for the project in PyCharm on Windows in order to work through the process in the book, but I am unable to install the packages from the requirements.txt file, specifically the PyYAML==5.4.1 and the uvloop==0.15.2 packages which say they are not supported on Windows.  It is also throwing additional errors from the install process when I try to install packages individually (see below).  Am I doing something wrong or is this book just well past its best-before expiration date...

Any assistance would be appreciated!  
Mary

Copy of a portion of the output from my terminal:
(web-apis-with-python) PS C:\Users\mbachjack\Sandbox\My_Pycharm\web-apis-with-python\web-apis-with-python> pip install wcwidth==0.2.5
Collecting wcwidth==0.2.5
  Using cached wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Installing collected packages: wcwidth
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pyecharts 2.0.3 requires jinja2, which is not installed.
Successfully installed wcwidth-0.2.5
(web-apis-with-python) PS C:\Users\mbachjack\Sandbox\My_Pycharm\web-apis-with-python\web-apis-with-python> pip install Jinja2==3.0.1
Collecting Jinja2==3.0.1
  Using cached Jinja2-3.0.1-py3-none-any.whl (133 kB)
Collecting MarkupSafe>=2.0 (from Jinja2==3.0.1)
  Obtaining dependency information for MarkupSafe>=2.0 from https://files.pythonhosted.org/packages/84/a8/c4aebb8a14a1d39d5135eb8233a0b95831cdc42c4088358449c3ed657044/MarkupSafe-2.1.3-cp310-cp310-win_amd64.whl.metada
ta
  Downloading MarkupSafe-2.1.3-cp310-cp310-win_amd64.whl.metadata (3.1 kB)
Using cached MarkupSafe-2.1.3-cp310-cp310-win_amd64.whl (17 kB)
Installing collected packages: MarkupSafe, Jinja2
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
altair 4.2.2 requires entrypoints, which is not installed.
altair 4.2.2 requires jsonschema>=3.0, which is not installed.
altair 4.2.2 requires numpy, which is not installed.
altair 4.2.2 requires toolz, which is not installed.
pydeck 0.8.0 requires numpy>=1.16.4, which is not installed.
torch 2.0.0 requires networkx, which is not installed.
torch 2.0.0 requires typing-extensions, which is not installed.
torchvision 0.15.1 requires numpy, which is not installed.
torchvision 0.15.1 requires requests, which is not installed.
ydata-profiling 4.1.2 requires numpy<1.24,>=1.16.0, which is not installed.
ydata-profiling 4.1.2 requires PyYAML<6.1,>=5.0.0, which is not installed.
ydata-profiling 4.1.2 requires requests<2.29,>=2.24.0, which is not installed.
ydata-profiling 4.1.2 requires seaborn<0.13,>=0.10.1, which is not installed.
ydata-profiling 4.1.2 requires tqdm<4.65,>=4.48.2, which is not installed.
Successfully installed Jinja2-3.0.1 MarkupSafe-2.1.3
(web-apis-with-python) PS C:\Users\mbachjack\Sandbox\My_Pycharm\web-apis-with-python\web-apis-with-python> pip install wcwidth==0.2.5
Requirement already satisfied: wcwidth==0.2.5 in c:\users\mbachjack\appdata\local\mambaforge\envs\web-apis-with-python\lib\site-packages (0.2.5)
(web-apis-with-python) PS C:\Users\mbachjack\Sandbox\My_Pycharm\web-apis-with-python\web-apis-with-python> pip install websockets==9.1
Collecting websockets==9.1
  Using cached websockets-9.1-cp310-cp310-win_amd64.whl
Installing collected packages: websockets
Successfully installed websockets-9.1
(web-apis-with-python) PS C:\Users\mbachjack\Sandbox\My_Pycharm\web-apis-with-python\web-apis-with-python> pip instal Werkzeug==2.0.1
ERROR: unknown command "instal" - maybe you meant "install"
(web-apis-with-python) PS C:\Users\mbachjack\Sandbox\My_Pycharm\web-apis-with-python\web-apis-with-python> pip install Werkzeug==2.0.1
Collecting Werkzeug==2.0.1
  Using cached Werkzeug-2.0.1-py3-none-any.whl (288 kB)
Installing collected packages: Werkzeug
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tensorboard 2.12.2 requires markdown>=2.6.8, which is not installed.
tensorboard 2.12.2 requires numpy>=1.12.0, which is not installed.
tensorboard 2.12.2 requires requests<3,>=2.21.0, which is not installed.
tensorboard 2.12.2 requires tensorboard-plugin-wit>=1.6.0, which is not installed.
tensorflow-intel 2.12.0 requires flatbuffers>=2.0, which is not installed.
tensorflow-intel 2.12.0 requires libclang>=13.0.0, which is not installed.
tensorflow-intel 2.12.0 requires numpy<1.24,>=1.22, which is not installed.
tensorflow-intel 2.12.0 requires packaging, which is not installed.
tensorflow-intel 2.12.0 requires six>=1.12.0, which is not installed.
tensorflow-intel 2.12.0 requires typing-extensions>=3.6.6, which is not installed.
tensorflow-intel 2.12.0 requires wrapt<1.15,>=1.11.0, which is not installed.
Successfully installed Werkzeug-2.0.1

**Comment by rehanhaider** [comment]
You have to be a bit more specific in terms of what the problem is. 

I.e. which exercise, what was the original code or link to the solution file that I've provided. Otherwise I'm trouble Python package issues.
